objects.py

```diff
@@ -0,0 +1,288 @@
+
+    @property
+    def price(self) -> float:
+        price = self.annual_energy * self.unit_cost
+        return price
+
+    @property
+    def wave_damping(self) -> float:
+        damping = self.capture_width_ratio_dict[self.wec_type]
+        return damping
+
+    @property
+    def capture_width_ratio(self) -> float:
+        capture_width_ratio = self.capture_width_ratio_dict[self.wec_type]
+        return capture_width_ratio
+
+class Wave:
+    def __init__(self, Hs: float, T: float) -> None:
+        self.Hs = Hs
+        self.T = T
+        self.rho = 1000
+        self.g = 9.81
+    
+    @property
+    def power(self) -> float:
+        P_wave = 1/32 * 1/pi * self.rho * self.g**2 * self.Hs**2 * self.T  #for regular wave
+        return P_wave
+
+class Fish:
+    def __init__(self, F_f, F_p, F_c, A_f, A_p, A_c, 
+                 O_f, O_p, O_c, C_f, C_p, C_c, 
+                 P_f, P_p, tau, loss_rate, harvest_weight, 
+                 O2_min, U_min, U_max, temp_min, temp_max, 
+                 salinity_min, salinity_max) :
+               
+        self.F_f = F_f
+        self.F_p = F_p
+        self.F_c = F_c
+
+        self.A_f = A_f
+        self.A_p = A_p
+        self.A_c = A_c
+        
+        self.O_f = O_f
+        self.O_p = O_p
+        self.O_c = O_c
+        
+        self.C_f = C_f
+        self.C_p = C_p
+        self.C_c = C_c
+        
+        self.P_f = P_f
+        self.P_p = P_p
+        self.tau = tau
+        self.loss_rate = loss_rate
+        self.harvest_weight = harvest_weight
+        
+        self.O2_min = O2_min
+        self.U_min = U_min
+        self.U_max = U_max
+        self.temp_min = temp_min
+        self.temp_max = temp_max
+        self.salinity_min = salinity_min
+        self.salinity_max = salinity_max
+        
+        self.time_i = []
+        self.W_i = []
+        self.W_dot_i = []
+        self.DO2_p_i = []
+        self.DO2_f_i = []
+        self.DO2_c_i = []
+        self.DO2_i = []
+        self.carrying_capacity_i = []
+    
+    def integrand(self, t):
+        return math.exp(self.temp * self.tau)
+    
+    def DO2(self, temp) -> float:
+        # specific energy content of feed
+        delta = self.F_f * self.C_f + self.F_p * self.C_p + self.F_c * self.C_c
+
+        # specific energy content of fish
+        C_f_star = 0.85 * self.C_p * self.P_p + self.C_f * self.P_f
+
+        # fraction of food energy contributed by progeins, fat, and carbs
+        E_p = self.F_p * self.C_p / delta
+        E_f = self.F_f * self.C_f / delta
+        E_c = self.F_c * self.C_c / delta
+
+        # metabolizable energy content of food
+        FL = (1-self.A_p) * E_p + (1-self.A_f) * E_f + (1-self.A_c) * E_c
+        BC = 0.3 * self.A_p * E_p + 0.05 * (self.A_f * E_f + self.A_c * E_c)
+        eps = 1 - FL - BC
+        eps_star = eps - 0.15 * self.F_p * self.C_p * self.A_p / delta
+
+        # Water temperature as a function of time
+        time_i = np.linspace(1, 365, 365) 
```

Comment:
for integers, `np.arange` is considered best practice

Reply:
Updated.